src/macro.jl

```diff
@@ -154,3 +155,29 @@ function _check_allocs_macro(ex::Expr, mod::Module, source::LineNumberNode; igno
         $(wrapper_fn)
     end
 end
+
+"""
+    @check_allocs_call ignore_throw=true (function call)
+
+Calls `check_allocs(func, types; ignore_throw)` where `types` is a tuple of the types of the arguments
+passed to `func` in the function call expression that `@check_allocs_call` is applied to.
+For example, `@check_allocs_call 1 + 2.0` is equivalent to `check_allocs(+, (Int, Float64))`.
+"""
+macro check_allocs_call(ex...)
```

Comment:
I think this can be a method/variant of `@check_allocs`. There is precedent for that with e.g. `@timeit` from [TimerOutputs.jl](https://github.com/KristofferC/TimerOutputs.jl): when applied to a function definition expression it enables timings for all calls, when applied to a call expression it times just that call.

Reply:
good idea, I wasn't aware this was being done elsewhere but seems relatively straightforward to explain, with no specific edge cases I could think of